renku/core/models/workflow/converters/cwl.py

```diff
@@ -279,6 +279,34 @@ def _convert_step(step, tmpdir, basedir, filename=None):
         for argument in step.arguments:
             tool_object.inputs.append(CWLConverter._convert_argument(argument))
 
+        workdir_req.listing.append(
+            cwlgen.InitialWorkDirRequirement.Dirent(
+                entry="$(inputs.input_renku_metadata)", entryname=".renku", writable=False
+            )
+        )
+        tool_object.inputs.append(
+            cwlgen.CommandInputParameter(
+                "input_renku_metadata",
+                param_type="Directory",
+                input_binding=None,
+                default={"path": os.path.abspath(os.path.join(basedir, ".renku")), "class": "Directory"},
+            )
+        )
+
+        workdir_req.listing.append(
+            cwlgen.InitialWorkDirRequirement.Dirent(
+                entry="$(inputs.input_git_directory)", entryname=".git", writable=False
```

Comment:
Did you check what the performance implication is of copying over the `.git` directory for each workflow rerun? Especially on large projects.

Reply:
It creates only a symlink and does not do an actual copy. This is not needed once we can load metadata without traversing git history. I'll add a TODO to remove when https://github.com/SwissDataScienceCenter/renku-python/issues/1043 is done.